xbmc/windowing/X11/WinSystemX11GLContext.cpp

```diff
@@ -177,18 +183,12 @@ bool CWinSystemX11GLContext::DestroyWindow()
 
 XVisualInfo* CWinSystemX11GLContext::GetVisual()
 {
-  GLint att[] =
-  {
-    GLX_RGBA,
-    GLX_RED_SIZE, 8,
-    GLX_GREEN_SIZE, 8,
-    GLX_BLUE_SIZE, 8,
-    GLX_ALPHA_SIZE, 8,
-    GLX_DEPTH_SIZE, 24,
-    GLX_DOUBLEBUFFER,
-    None
-  };
-  return glXChooseVisual(m_dpy, m_nScreen, att);
+  int count;
+  XVisualInfo* visuals = nullptr;
+
+  visuals = XGetVisualInfo(m_dpy, 0, NULL, &count);
```

Comment:
you have to select a suitable visual here. create a template, then iterate through the number of returned and select the best matching. use XFree when finished
ref: https://www.x.org/archive/X11R7.5/doc/man/man3/XVisualInfo.3.html

Reply:
Is that really needed? after the context is created a new suitable visual is selected either for glx or egl. 

https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/X11/GLContextEGL.cpp#L266
https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/X11/GLContextGLX.cpp#L142

Reply:
I don't think you understand the situation correctly. visual is requred for creating the window. this is before any gl context is created. 
When we create the GL/EGL context, we just verify if the visual is suitable.

Reply:
I don't think that's correct. It looks to my like it will try to find and use a suitable visual for each context.
https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/X11/GLContextEGL.cpp#L89-L136
https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/X11/GLContextGLX.cpp#L52-L128

Reply:
Right, visual is used to get i.e. eglConfig but the visual used here should match the visual used for creating the window. 
```
For EGL window and pixmap surfaces, a suitable native window or pixmap with a matching native visual must be created first.
```
https://www.khronos.org/registry/EGL/sdk/docs/man/html/eglIntro.xhtml

We don't want to make this working by chance if the first visual returned matches. Further I am not sure if there are any internal convertions required if both visuals do not match.

Reply:
what about using XMatchVisualInfo instead (https://tronche.com/gui/x/xlib/utilities/XMatchVisualInfo.html)

something like
```
  XVisualInfo *visual = new XVisualInfo();
  XMatchVisualInfo(m_dpy, m_nScreen, 24, TrueColor, visual);
  return visual;
```
This will match a visual with the correct depth and color that we want.

Reply:
this is not what we want. we want exaxtly the same parameters we use for glx/egl. the solution is rather simple: submit a template with required parameters to XGetVisualInfo